### ISSUE TITLE
Perform Lighting Calculations in Vertex Shader

### ISSUE THREAD

**Issue by marauder00** [ProtoArt/spritec#8]
[reposting from pull request # 1]
Put the basic diffuse/ambient color code in the vertex shader. There are less vertices than fragments, so less computations will be performed.

Usually this isn't done in sophisticated rendering engines because you lose fidelity when you're interpolating colors between entire vertices vs pixel fragments. HOWEVER because pixel art is meant to be seen at very low resolutions, chances are, the fidelity loss is minimal.

**Comment by sunjay** [comment]
This is a good idea, but low priority for the timebeing, so I have unassigned the task. Let's focus on getting better outlines (#7) before we start optimizing things. The shader is also going to change a lot when I add lighting support (#28), so we should hold off on making decisions like this until we're more feature complete.
